File: agent/commands-generator-api/src/main/java/org/matilda/commands/PluginDependenciesModule.java
package org.matilda.commands;

import dagger.Module;
import dagger.Provides;

import java.util.Random;

@Module
public class PluginDependenciesModule {
    private final PluginDependencies mPluginDependencies;

    public PluginDependenciesModule(PluginDependencies pluginDependencies) {
        mPluginDependencies = pluginDependencies;
    }

    @Provides
    CommandRegistryManager commandRegistryManager() {
        return mPluginDependencies.commandRegistryManager;
    }

    @Provides
    CommandRunner commandRunner() {
        return mPluginDependencies.commandRunner;
    }

    @Provides
    Random random() {
        return mPluginDependencies.random;
    }
}

File: agent/commands-generator-api/src/main/java/org/matilda/commands/PluginDependenciesModule.kt
package org.matilda.commands

import dagger.Module
import dagger.Provides

@Module
class PluginDependenciesModule(private val mPluginDependencies: PluginDependencies) {
    @Provides
    fun commandRegistryManager() = mPluginDependencies.commandRegistryManager

    @Provides
    fun commandRunner() = mPluginDependencies.commandRunner

    @Provides
    fun random() = mPluginDependencies.random
}
